src/main/java/com/teragrep/pth_06/planner/walker/conditions/IndexCondition.java

```diff
@@ -71,22 +71,38 @@ public Condition condition() {
             condition = StreamDBClient.GetArchivedObjectsFilterTable.directory
                     .like(value.replace('*', '%').toLowerCase());
         }
-        if (operation.equalsIgnoreCase("NOT_EQUALS")) {
+        if ("NOT_EQUALS".equalsIgnoreCase(operation)) {
             condition = condition.not();
         }
         return condition;
     }
 
     @Override
     public boolean equals(final Object object) {
-        if (this == object)
+        if (this == object) {
             return true;
-        if (object == null)
+        }
+        if (object == null) {
             return false;
-        if (object.getClass() != this.getClass())
+        }
+        if (object.getClass() != this.getClass()) {
             return false;
+        }
         final IndexCondition cast = (IndexCondition) object;
         return this.streamQuery == cast.streamQuery && this.value.equals(cast.value)
                 && this.operation.equals(cast.operation);
     }
+
+    @Override
+    public int hashCode() {
+        int result = 17;
```

Comment:
hashcode

Reply:
Solved in commit 4d4e0023426356b05c592147f65933f69ba5dfa3